### ISSUE TITLE
Rule suggestion: opposite to `no-else-return`

### ISSUE THREAD

**Issue by OliverJAsh** [eslint/eslint#10634]
**Please describe what the rule should do:**
The opposite to [`no-else-return`].

Examples of incorrect code for this rule:

``` js
function foo() {
    if (x) {
        return y;
    }

    return z;
}
```

Examples of correct code for this rule:

``` js
function foo() {
    if (x) {
        return y;
    } else {
        return z;
    }
}
```

Justification: a function is easier to scan if all the code paths are indented. The indentation makes it clear that the function may be returning somewhere above—without the indentation from branching in an `if` statement, this might not be clear, especially in a large function.

``` js
if (x) {
  return true;
}

// 20 lines down, or so…
// Why is my code not executing?!?!
console.log('Hello, World!');
```

``` js
if (x) {
  return true;
} else {
  // Indentation makes it clear that this is a separate branch
  console.log('Hello, World!');
}
```

**What category of rule is this? (place an "X" next to just one item)**

[x] Enforces code style
[ ] Warns about a potential error
[x] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
function foo() {
    if (x) {
        return y;
    }

    return z;
}

function foo() {
    if (error) {
        return 'It failed';
    }

    if (loading) {
        return "It's still loading";
    }
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

For the same reason [`no-else-return`] is included. In my experience, the style suggested here is just as popular, especially in functional programming.

[`no-else-return`]: https://eslint.org/docs/rules/no-else-return

**Comment by kaicataldo** [comment]
I don't think this warrants a separate rule. If we did want to add this to core, it would make more sense to me as an option for the `no-else-return` (something equivalent to `"always"` or `"never"`, though I think the double negative is confusing in this case).
